packages/flutter/lib/src/material/selection_area.dart

```diff
@@ -63,6 +64,9 @@ class SelectionArea extends StatefulWidget {
   /// If it is null, the platform specific selection control is used.
   final TextSelectionControls? selectionControls;
 
+  /// Called when the selected content changes.
+  final Function(String)? onSelectionChange;
```

Comment:
This should probably return SelectedContent

plainText is just one representation, in the future we will probably support html for rich content.

and you should use typedef to define the function signature into a type.